q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,332 @@
+        of the variables rules, or during the application the rules.
+
+    Methods
+    -------
+    check_variables_rules
+        Fills `rules` attribute with correctly formatted rules
+        for the current variable, or raise user-defined exception.
+    _check_variables_rules_dir
+        Performs checks on rules paths input. Either raise Exceptions,
+        or return the list `variables_rules_files` containing the paths
+        to the existing yml rules files.
+    parse_variables_rules
+        Parse the yml rules files, one variable at a time and returns the
+        dict `variables_rules` that for each variable (keys) leads to an
+        instance of the `Rules()` class. This class performs rule-formatting
+        checks and fills a rules dict structure.
+
+    Raises
+    ------
+    IOError
+        If directory passed to --p-rules-dir does not exist.
+    FileNotFoundError
+        If directory passed to --p-rules-dir contains no .yml file.
+
+    """
+
+    def __init__(self) -> None:
+        """Initialize the class instance for the collection of the set
+        of rules associated with all the metadata variables. The instantiated class will be
+        given a `variable` attribute corresponding to the variable name
+        as string, as well as a `rules` attribute collecting the actual
+        rules in a dictionary.
+
+        """
+        self.warnings = WarningsCollection()
+        self.errors = ErrorsCollection()
+        self.variables_rules = {}
+
+    def parse_variables_rules(self, variables_rules_files: list) -> None:
+        """Parse the variables yaml rues files one by one.
+        At this point, the attribute `variables_rules` will
+        have for each variable (keys), an instance of the
+        Rules() class, initiated with an empty default data
+        structure `rules` and the yet unchecked `parsed_rules`.
+
+        Parameters
+        ----------
+        variables_rules_files : list
+            Paths to the yaml rules files.
+
+        """
+        for variable_rules_fp in variables_rules_files:
+            variable = splitext(basename(variable_rules_fp))[0]
+            # instantiate Rules() class to get variable's yaml rules
+            self.variables_rules[variable] = Rules(variable_rules_fp)
+
+    def check_variables_rules(self, focus: list) -> None:
+        """This method fills the rules attribute with the correctly
+        formatted rules for the current variable, or or raise user-
+        defined exception.
+
+        Parameters
+        ----------
+        focus : list
+            Metadata variables that have rules.
+
+        """
+        for variable in focus:
+            variable_rules = self.variables_rules[variable]
+            for rule, rule_value in variable_rules.parsed_rules.items():
+                check_rule(variable, variable_rules.rules, rule, rule_value)
+
+    @staticmethod
+    def _check_variables_rules_dir(variables_rules_dir: str) -> list:
```

Comment:
```suggestion
    def check_variables_rules_dir(variables_rules_dir: str) -> list:
```

Reply:
accounted for - will be part of upcoming commit